index.js

```diff
@@ -0,0 +1,156 @@
+    // bot was invited to the channel :(
+
+    // looks like self.id should always be present - https://api.slack.com/methods/rtm.start
+    botID = rtmStartData.self.id.toLowerCase();
+    console.log(`Logged in as ${rtmStartData.self.name}(${botID}) of team ${rtmStartData.team.name}, but not yet connected to a channel`);
+});
+
+let rtmConnectionOpened = false;
+
+rtm.on(RTM_EVENTS.MESSAGE, function handleRtmMessage(message) {
+
+    if (!message.text) {
+        return;
+    }
+
+    // forgive trailing spaces and ignore case
+    const normalizedMessage = message.text.trim().toLowerCase();
+    let now = false;
+    let tomorrow = false;
+
+    // only respond to messages where we are mentioned
+    if (normalizedMessage.indexOf(`<@${botID}>`) === -1) {
+        return;
+    }
+
+    // private channel ID isn't available on start, so respond on the channel
+    // we're mentioned on.
+    botChannel = message.channel;
+
+    if (normalizedMessage.indexOf('weather now') !== -1) {
+        now = true;
+    }
+
+    if (normalizedMessage.indexOf('weather tomorrow') !== -1) {
+        tomorrow = true;
+    }
+
+    if (!now && !tomorrow) {
+        if (rtmConnectionOpened) {
+            if (normalizedMessage.indexOf('help') !== -1) {
+                rtm.sendMessage(`<@${message.user}>, try @surry-interview 'weather now' or 'weather tomorrow'`, botChannel);
+            } else {
+                rtm.sendMessage(`<@${message.user}>, I did not understand that command, try @surry-interview 'weather now' or 'weather tomorrow'`, botChannel);
+                console.log('Ignoring unknown command: ' + normalizedMessage.substring(0, 20) + '...');
+            }
+        }
+        return;
+    }
+
+    if (now) {
+        util.getCurrentWeatherReport().then((json) => {
+            if (rtmConnectionOpened) {
+                if (json.currently) {
+                    let message = util.generateCurrentSummary(json.currently);
+                    rtm.sendMessage(message, botChannel);
+
+                    const now = new Date().getTime();
+                    // This url can be used to grab a frame from a DDOT (DC) traffic cam!
+                    // This can be used to give a rough visual of the weather. The camera is
+                    // at the intersection of 14th St and Independence St NW. Looks like the current
+                    // time in milliseconds since 1970 needs to be passed in to get the latest images.
+                    // The RTM client doesn't allow sending attachments (see https://api.slack.com/rtm),
+                    // but the Slack URL unfurling functionality automatically includes the traffic cam frame!
+                    // See http://app.ddot.dc.gov/ for more info.
+                    const trafficCamUrl = `http://ie.trafficland.com/v1.0/200146/full?system=ddot&pubtoken=b4c0b819f66741b99fdb17c963f47e3bafd09b494019ee9624f1d38641ebc6bf&refreshRate=2000&t=${now}`
```

Comment:
I wanted to see if I could include an attachment that would show the satellite imagery for a region when someone issued the *weather now* command, but I didn't have any luck with the APIs I found. I also found out the RTM client can't send attachments with messages.

I ended up relying on the URL unfurling on the Slack server-side, and found a link that returns images to give a visual representation of the weather.